server/grpc_service.go

```diff
@@ -539,18 +560,14 @@ func (s *GrpcServer) Tso(stream pdpb.PD_TsoServer) error {
 				return errors.WithStack(err)
 			}
 
-			if errCh == nil {
-				doneCh = make(chan struct{})
-				defer close(doneCh) // nolint
-				errCh = make(chan error)
-			}
-
 			tsoRequest := tsoutil.NewPDProtoRequest(forwardedHost, clientConn, request, stream)
-			s.tsoDispatcher.DispatchRequest(ctx, tsoRequest, s.pdProtoFactory, doneCh, errCh, s.tsoPrimaryWatcher)
+			// don't pass a stream context here as dispatcher serves multiple streams
```

Comment:
Will these be closed if we disable the proxy?

Reply:
I was thinking whether there is a resource leak after scaling in and out repeatedly.

Reply:
> No, but it is at most one connection to each PD, which won't be re-open after restrat. 
Every client stream is independent, so the stream may have count(TIDB X PD)